### ISSUE TITLE
ZenDesk App

### ISSUE THREAD

**Issue by jaswsinc** [websharks/wp-kb-articles#96]
It would be great to have a ZenDesk app powered by WPKBA.

- [ ] Bring KB articles into ZenDesk sidebar.
- [ ] Fast AJAX search box with one-click insertion into ZenDesk response.
- [ ] Automatic KB article suggestions based on ticket content analysis.
- [ ] If integrated with GitHub (i.e., Markdown), provide agents with the option to insert the full article body into the response instead of a link—should they choose to do so.
- [ ] Make it possible to map customer service Depts., Tags or Custom Fields to a specific instance of WPKBA; i.e., a specific instance of WordPress.
- [ ] Build the app itself into the WordPress plugin if possible.

Referencing: https://developer.zendesk.com/apps/docs/agent/introduction

cc @raamdev

**Comment by raamdev** [comment]
`str_replace('ZenDesk', 'Zendesk', $jason);`

**Comment by raamdev** [comment]
> It would be great to have a Zendesk app powered by WPKBA.

Wow, that would be awesome!

Along these same lines, I was going to integrate a list of our KB Articles into Typinator so that we could easily search our KB Articles using the same Quick Search interface that we use for Menu Paths. However, the ability to _suggest_ KB Articles based on the ticket content would be just awesome!

**Comment by jaswsinc** [comment]
@raamdev writes...
> str_replace('ZenDesk', 'Zendesk', $jason);

That's awesome. ahaha

**Comment by jaswsinc** [comment]
@raamdev writes...
> I was going to integrate a list of our KB Articles into Typinator so that we could easily search our KB Articles using the same Quick Search interface that we use for Menu Paths.

Cool! Even if we do the Zen**d**esk app sooner rather than later (likely), I still think Typinator shortcuts would be really useful. It might even encourage me to hit the community forums more often.

**Comment by raamdev** [comment]
> I still think Typinator shortcuts would be really useful

Yep, I agree. I see the Typinator KB Article shortcuts being _really_ useful, because the more familiar we (as support reps) get with our own KB and knowing which articles should be used to reply to which questions, we'll start to build muscle memory around finding and referencing those KB Articles using the Typinator Quick Search. Also, Typinator works from _anywhere_, as opposed to only inside Zendesk, so as you said it would be useful for the Community Forums too.
